### PR TITLE
SE-10166: Added system env macros - can determine current system environment

### PR DESCRIPTION
**Pull request by marvinwoop**
To determine the current system environment in templates, we need this SystemHelper.

### CODE REVIEW

File: src/main/java/sirius/web/util/SystemHelper.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        }
+    }
+
+    /**
+     * Returns the value of the given environment variable.
+     *
+     * @param varName the environment variable to look up
+     * @return the value of the environment variable
+     */
+    public String getEnvironmentVariable(String varName) {
```
Comment (andyHa) [inline src/main/java/sirius/web/util/SystemHelper.java:53]: this is VERY DANGEROUS! why is that needed?

File: src/main/java/sirius/web/util/SystemHelper.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import sirius.web.security.HelperFactory;
+import sirius.web.security.ScopeInfo;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Provides static system functionality and system environment detection based on {@link sirius.kernel.Setup.Mode} as
+ * helper for in-template usage.
+ */
+public class SystemHelper {
```
Comment (andyHa) [inline src/main/java/sirius/web/util/SystemHelper.java:23]: we could also encapsulate all this in a isProd() / isStaging() / isDev() macro which is probably easier to use anyway... (?)

**Review by andyHa — CHANGES_REQUESTED** [review]

File: src/main/java/sirius/web/util/SystemHelper.java (hunk 1 above)
Comment (marvinwoop) [reply src/main/java/sirius/web/util/SystemHelper.java:53]: This is not really needed and will therefore be removed.

File: src/main/java/sirius/web/util/SystemHelper.java (hunk 2 above)
Comment (marvinwoop) [reply src/main/java/sirius/web/util/SystemHelper.java:23]: This will be rebuilt into a macro as suggested.

File: src/main/java/sirius/tagliatelle/macros/IsProductiveMacro.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import sirius.tagliatelle.rendering.LocalRenderContext;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+/**
+ * This macro determines if the current system environment is in production {@link sirius.kernel.Setup.Mode}.
+ */
+@Register
+public class IsProductiveMacro implements Macro {
```
Comment (andyHa) [inline src/main/java/sirius/tagliatelle/macros/IsProductiveMacro.java:25]: extract common superclass? so that there is only a single getMode() left to implement...

File: src/main/java/sirius/tagliatelle/macros/IsProductiveMacro.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    @Override
+    public Object eval(LocalRenderContext ctx, Expression[] args) {
+        return Sirius.isProd();
+    }
+
+    @Override
+    public boolean isConstant(Expression[] args) {
+        return false;
```
Comment (andyHa) [inline src/main/java/sirius/tagliatelle/macros/IsProductiveMacro.java:45]: this can't ever change -> true

Comment (marvinwoop) [reply src/main/java/sirius/tagliatelle/macros/IsProductiveMacro.java]: 👍  changed that as suggested in base class.

**Review by andyHa — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

File: src/main/java/sirius/tagliatelle/macros/IsProductiveMacro.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Override
+    public String getDescription() {
+        return "Returns true if system environment in production mode, false otherwise.";
+    }
+
+    @Nonnull
+    @Override
+    public String getName() {
+        return "isProductive";
```
Comment (sabieber) [inline src/main/java/sirius/tagliatelle/macros/IsProductiveMacro.java:37]: Why use a different naming here that the sirius version and the enum value?

**Review by sabieber — CHANGES_REQUESTED** [review]

File: src/main/java/sirius/tagliatelle/macros/IsProductiveMacro.java (hunk 5 above)
Comment (marvinwoop) [reply src/main/java/sirius/tagliatelle/macros/IsProductiveMacro.java]: thanks, changed the naming. 👍

**Review by sabieber — APPROVED** [review]
